business/workloads.go

```diff
@@ -604,15 +616,8 @@ func fetchWorkloads(ctx context.Context, layer *Layer, namespace string, labelSe
 	// Pods are always fetched
 	go func() {
 		defer wg.Done()
-
-		// Check if namespace is cached
-		// Namespace access is checked in the upper caller
 		var err error
-		if IsNamespaceCached(namespace) {
-			pods, err = kialiCache.GetPods(namespace, labelSelector)
-		} else {
-			pods, err = layer.k8s.GetPods(namespace, labelSelector)
-		}
+		pods, err = layer.k8s.GetPods(namespace, labelSelector)
```

Comment:
So `layer.k8s` will use the `cachingClient` when the kube cache is enabled which is just a different implementation of the client interface that uses the cache when the cache is supported for that method call like `GetPods`. We don't need the `cachingClient` and can instead call the kiali cache directly in the business layer as it's done here but imo it makes it easier to transition these services over to just using the cache.

Reply:
OK, now I got it.
Probably I was confused and I thought that `layer.k8s` was always the usual kubeclient giving me the idea that the cache was going to be bypassed. But I see that in `layer.go` it will be either the `K8sClient` or the `cachingClient`. So, all good.